src/core/egg-herbie.rkt

```diff
@@ -3,15 +3,64 @@
 (require egg-herbie)
 (require ffi/unsafe ffi/unsafe/define)
 (require "../syntax/rules.rkt" "../syntax/sugar.rkt" "../syntax/syntax.rkt" "../syntax/types.rkt"
-         "../common.rkt" "../errors.rkt" "../timeline.rkt")
+         "../common.rkt" "../errors.rkt" "../timeline.rkt" "../alternative.rkt")
 
 (module+ test (require rackunit))
 
 (provide make-egraph egraph-add-expr egraph-run-rules
          egraph-get-simplest egraph-get-variants
          egraph-get-proof egraph-is-unsound-detected
          rule->egg-rules expand-rules get-canon-rule-name
-         remove-rewrites)
+         remove-rewrites run-egg make-egg-descriptor
+         (struct-out proof-input) (struct-out egraph-input))
+
+(struct egraph-input (exprs rules num-variants terms iter-limit node-limit const-folding) #:transparent)
+(struct proof-input (start end) #:transparent)
```

Comment:
I'm not a huge fan of this API. For example, I think `num-variants` shouldn't actually affect how the egraph is run, so does it need to be part of the `egraph-input`? Also, I'm not sure why we need `proof-input` as a separate struct. Also it is a bit of an ugly API that you use `run-egg` for both getting simplified versions and proofs. (Since all callers need exactly one or the other.)

Reply:
For the num-variants portion, it directly ties into the second question. I think Zach, Oliver and Brett wanted a general singular function that abstracts over both `simplified` and `rr`, and the best way for us to differentiate which variant to return was through the num-variants of the instance of the `egraph-input struct`. We can revert it back, but I think we wanted it this way.

Reply:
Sorry about that I think I misunderstood Pavel's initial comment then. That should be fine to fix then.

Reply:
I'm confused about the way to go here. Should we keep `num-variants`? I agree with removing `proof-input` as a separate struct.

Reply:
There should be a single `egraph-input` struct with *all of the inputs that affect the e-graph that gets built*. But num-variants doesn't affect the egraph that gets built, just what expressions are extracted out of it, so that shouldn't go in the command.